src/test/scala/chiselTests/naming/SuggestNameSpec.scala

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests.naming
+
+import chisel3._
+import chisel3.aop.Select
+import chisel3.stage.{ChiselGeneratorAnnotation, ChiselStage}
+import chisel3.experimental.{dump, noPrefix, prefix, treedump}
+import chiselTests.{ChiselPropSpec, Utils}
+import chisel3.stage.{DesignAnnotation, NoRunFirrtlCompilerAnnotation}
+import firrtl.options.{Dependency, Phase, PhaseManager}
+import firrtl.options.phases.DeletedWrapper
+
+class SuggestNameSpec extends ChiselPropSpec with Utils {
+  implicit val minimumMajorVersion: Int = 12
+
+  property("0. Calling suggestName 2x should be a runtime deprecation") {
+    class Test extends Module {
+      {
+        val wire = {
+          val x = WireInit(0.U(3.W)).suggestName("mywire")
+          dontTouch(x)
+        }
+        wire.suggestName("somethingElse")
+      }
+    }
+    val (log, _) = grabLog(ChiselStage.emitVerilog(new Test()))
+    log should include(
+      "Calling suggestName(\"somethingElse\"), when already called with \"mywire\", will become an error in Chisel 3.6"
+    )
+  }
+
+  property("1. Calling suggestName outside of a Builder context should be a runtime deprecation") {
+    class Test extends Module {
+      val wire = {
+        val x = WireInit(0.U(3.W))
+        dontTouch(x)
+      }
+    }
+
+    // Nasty use of var, only for this test purpose. Don't do stuff like this!
+    var test: Test = null
+    ChiselStage.elaborate {
+      test = new Test
+      test
+    }
+    val (log, _) = grabLog {
+      test.wire.suggestName("somethingElse")
+    }
+    log should include("suggestName(\"somethingElse\") should only be called from a Builder context")
```

Comment:
@jackkoenig this test doesn't actually raise any runtime deprecations, what am I missing? I would assume I am not in a Builder context...

Reply:
I had to do a similar thing in https://github.com/chipsalliance/chisel3/pull/2550, check it out. It's awkward but this is a warning we'll add and backport, and then promptly delete.

Reply:
excellent, tried that and it worked, thanks!